### DISCUSSION TITLE
use `name` when defined as `__repr__` for Dataset and Model

### DISCUSSION THREAD

**Discussion by AdriMarteau** [Giskard-AI/giskard-oss#1723] [Ideas]
This can make some of the execution report more meaningful compare to, for instance: `<giskard.datasets.base.Dataset object at 0x145344890>`

**Comment by andreybavt** [comment]
we should also add `id` to the `__repr__`

**Reply by AdriMarteau** [reply]
I was thinking of doing:
```python
...
def __repr__(self):
   if self.name:
      return f"{self.name}-{self.id}"
   return super().__repr__()
```

**Comment by luca-martial** [comment]
@AdriMarteau this sounds good to us - would you be open to implementing this as an external contributor?

**Reply by AdriMarteau** [reply]
Sure! I'm getting more familiar with the code now 🙂
Let me have a look at this once my other PR is merged.

**Reply by luca-martial** [reply]
Awesome! Thank you for your contributions!

**Comment by AdriMarteau** [comment]
here is the corresponding PR: https://github.com/Giskard-AI/giskard/pull/1732
